src/DevantlerTech.KubernetesGenerator.Native/CronJobGenerator.cs

```
@@ -1,11 +1,72 @@
+using System.Collections.ObjectModel;
 using DevantlerTech.KubernetesGenerator.Core;
-using k8s.Models;
+using DevantlerTech.KubernetesGenerator.Native.Models;
 
 namespace DevantlerTech.KubernetesGenerator.Native;
 
 /// <summary>
-/// A generator for Kubernetes CronJob objects.
+/// A generator for Kubernetes CronJob objects using 'kubectl create cronjob' commands.
 /// </summary>
-public class CronJobGenerator : BaseKubernetesGenerator<V1CronJob>
+public class CronJobGenerator : BaseNativeGenerator<CronJob>
 {
+  static readonly string[] _defaultArgs = ["create", "cronjob"];
+
+  /// <summary>
+  /// Generates a cronjob using kubectl create cronjob command.
+  /// </summary>
+  /// <param name="model">The cronjob object.</param>
+  /// <param name="outputPath">The output path for the generated YAML.</param>
+  /// <param name="overwrite">Whether to overwrite existing files.</param>
+  /// <param name="cancellationToken">The cancellation token.</param>
+  /// <exception cref="ArgumentNullException">Thrown when model is null.</exception>
+  /// <exception cref="KubernetesGeneratorException">Thrown when required parameters are missing.</exception>
+  public override async Task GenerateAsync(CronJob model, string outputPath, bool overwrite = false, CancellationToken cancellationToken = default)
+  {
+    ArgumentNullException.ThrowIfNull(model);
+
+    var args = new ReadOnlyCollection<string>(
+      [.. _defaultArgs, .. AddArguments(model)]
+    );
+    string errorMessage = $"Failed to create cronjob '{model.Metadata?.Name}' using kubectl";
+    await RunKubectlAsync(outputPath, overwrite, args, errorMessage, cancellationToken).ConfigureAwait(false);
```

Do not use null coalescence for required values. And ensure name is not empty, that should throw an error.